jenkins-job-components/src/main/java/com/ofg/job/component/JobComponent.java

```diff
@@ -0,0 +1,20 @@
+package com.ofg.job.component;
+
+import javaposse.jobdsl.dsl.Job;
+
+/**
+ * Interface for configurations applicable on Job.
+ */
+public interface JobComponent<J extends Job> {
+
+    JobComponent<? extends Job> EMPTY = job -> {};
+    
+    default <NJ extends J> JobComponent<NJ> and(JobComponent<NJ> other) {
```

Comment:
Another thought: we could pull this off with just one operator, and a `configuring(Class<NJ>)` method. This way we could

```
SpecificConfigurer &
AbstractConf &
SpecConf //won't compile: & AnotherSpecConf
```
and if one needs to start with the abstact configurer, we can introduce a type-sepcifying factory method:

```
configuring(MultiJob) & //returns an empty JobComponent 
AbsConfigurer & 
SpecConfigurer &
AbsConfigurer
```